tests/api/test_connection.py

```diff
@@ -260,6 +280,30 @@ def function(*args, **kwargs):
         # A successful cancellation does not raise an exception
         connection.cancel_job("123")
 
+    def test_cancel_pending_job(self, connection, monkeypatch):
+        """Tests a job cancellation first is pending and then is successful by
+        using a mock server."""
+        # A custom `mock_return` that checks for expected arguments
+        def _mock_return(return_value):
+            def function(*args, **kwargs):
+                assert kwargs.get("json") == {"status": "cancelled"}
+                return return_value
+
+            return function
```

Comment:
Correct me if I'm wrong, but this mock doesn't seem to be used anywhere :thinking: And it seems that the same logic is covered in `MockCancelerServer.patch`, so maybe you meant to delete this?